README.md

```diff
@@ -1,7 +1,197 @@
+
+    I have 42 cucumbers in my belly
+
+The simplest Cucumber Expression that matches that text would be the text itself,
+but we can also write a more generic expression, with an `int` *output parameter*:
+
+    I have {int} cucumbers in my belly
+
+When the text is matched against that expression, the number `42` is extracted
+from the `{int}` output parameter and passed as an argument to the [step definition](https://cucumber.io/docs/cucumber/step-definitions).
+
+The following text would not match the expression:
+
+    I have 42.5 cucumbers in my belly
+
+This is because `42.5` has a decimal part, and doesn't fit into an `int`.
+Let's change the output parameter to `float` instead:
+
+    I have {float} cucumbers in my belly
+
+Now the expression will match the text, and the float `42.5` is extracted.
+
+## Parameter types
+
+Text between curly braces reference a *parameter type*. Cucumber comes with
+the following built-in parameter types:
+
+| Parameter Type  | Description |
+| --------------- | ----------- |
+| `{int}`         | Matches integers, for example `71` or `-19`. |
+| `{float}`       | Matches floats, for example `3.6`, `.8` or `-9.2`. |
+| `{word}`        | Matches words without whitespace, for example `banana` (but not `banana split`). |
+| `{string}`      | Matches single-quoted or double-quoted strings, for example `"banana split"` or `'banana split'` (but not `banana split`). Only the text between the quotes will be extracted. The quotes themselves are discarded. Empty pairs of quotes are valid and will be matched and passed to step code as empty strings. |
+| `{}` anonymous  | Matches anything (`/.*/`). |
+
+### Cucumber-JVM additions
+
+On the JVM, there are additional parameter types for `biginteger`, `bigdecimal`,
+`byte`, `short`, `long` and `double`.
+
+The anonymous parameter type will be converted to the parameter type of the step definition using an object mapper.
+Cucumber comes with a built-in object mapper that can handle most basic types. Aside from `Enum` it supports conversion
+to `BigInteger`, `BigDecimal`, `Boolean`, `Byte`, `Short`, `Integer`, `Long`, `Float`, `Double` and `String`.
+
+To automatically convert to other types it is recommended to install an object mapper. See [configuration](https://cucumber.io/docs/cucumber/configuration)
+to learn how.
+
+### Custom Parameter types
+
+Cucumber Expressions can be extended so they automatically convert
+output parameters to your own types. Consider this Cucumber Expression:
+
+    I have a {color} ball
+
+If we want the `{color}` output parameter to be converted to a `Color` object,
+we can define a custom parameter type in Cucumber's [configuration](https://cucumber.io/docs/cucumber/configuration).
+
+The table below explains the various arguments you can pass when defining
+a parameter type.
+
+| Argument      | Description |
+| ------------- | ----------- |
+| `name`        | The name the parameter type will be recognised by in output parameters.
```

Comment:
Are we using American or British English as our target language? (US color+recognize vs UK colour+recognise.)

Reply:
I think British English is the predominant language used in our docs, but I’m sure there are many inconsistencies.